examples/cosmos/chain_export_test.go

```diff
@@ -0,0 +1,118 @@
+package cosmos_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/strangelove-ventures/interchaintest/v8"
+	"github.com/strangelove-ventures/interchaintest/v8/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v8/ibc"
+	"github.com/strangelove-ventures/interchaintest/v8/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestJunoStateExport(t *testing.T) {
+	// SDK v45
+	CosmosChainStateExportTest(t, "juno", "v15.0.0", false)
+	// SDK v47
+	CosmosChainStateExportTest(t, "juno", "v16.0.0", true)
+}
+
+func CosmosChainStateExportTest(t *testing.T, name, version string, useNewGenesisCmd bool) {
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+
+	numVals := 1
+	numFullNodes := 0
+
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			Name:      name,
+			ChainName: name,
+			Version:   version,
+			ChainConfig: ibc.ChainConfig{
+				Denom:                  "ujuno",
+				UsingNewGenesisCommand: useNewGenesisCmd,
+			},
+			NumValidators: &numVals,
+			NumFullNodes:  &numFullNodes,
+		},
+	})
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	chain := chains[0].(*cosmos.CosmosChain)
+
+	ic := interchaintest.NewInterchain().
+		AddChain(chain)
+
+	ctx := context.Background()
+	client, network := interchaintest.DockerSetup(t)
+
+	require.NoError(t, ic.Build(ctx, nil, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: true,
+	}))
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	HaltChainAndExportGenesis(ctx, t, chain, nil, 3)
+}
+
+func HaltChainAndExportGenesis(ctx context.Context, t *testing.T, chain *cosmos.CosmosChain, relayer ibc.Relayer, haltHeight int64) {
+	timeoutCtx, timeoutCtxCancel := context.WithTimeout(ctx, time.Minute*2)
+	defer timeoutCtxCancel()
+
+	err := testutil.WaitForBlocks(timeoutCtx, int(haltHeight), chain)
+	require.NoError(t, err, "chain did not halt at halt height")
+
+	err = chain.StopAllNodes(ctx)
+	require.NoError(t, err, "error stopping node(s)")
+
+	state, err := chain.ExportState(ctx, int64(haltHeight))
+	require.NoError(t, err, "error exporting state")
+
+	appTomlOverrides := make(testutil.Toml)
+
+	appTomlOverrides["halt-height"] = 0
```

Comment:
Can you explain why this is necessary? `halt-height` defaults to `0` yea? Does it get overwritten when a node is stopped?

Reply:
Oops! not necessary. Has no effect for this test. Removed in 4b6cadaca4a55f73848943d16e8e6bd8f94bd0c9